### PR TITLE
Project overview data layer

### PR DESCRIPTION
**Pull request by arnekuchenbecker**

### CODE REVIEW

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/daos/AllergenDAO.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/*
+ * KitchenPlanerApp is the android app frontend for the KitchenPlaner, a tool
+ * to cooperatively plan a meal plan for a campout.
+ * Copyright (C) 2023  Arne Kuchenbecker, Antonia Heiming
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/datalayer/daos/AllergenDAO.kt:4]: Update Licence

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/entities/PersonNumberChangeEntity.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/*
+ * KitchenPlanerApp is the android app frontend for the KitchenPlaner, a tool
+ * to cooperatively plan a meal plan for a campout.
+ * Copyright (C) 2023  Arne Kuchenbecker, Antonia Heiming
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/datalayer/entities/PersonNumberChangeEntity.kt:4]: Change Licence

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/entities/RecipeProjectMealEntity.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/*
+ * KitchenPlanerApp is the android app frontend for the KitchenPlaner, a tool
+ * to cooperatively plan a meal plan for a campout.
+ * Copyright (C) 2023  Arne Kuchenbecker, Antonia Heiming
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/datalayer/entities/RecipeProjectMealEntity.kt:4]: Change Licence

File: app/src/main/java/com/scouts/kitchenplaner/exceptions/DuplicatePrimaryKeyException.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/*
+ * KitchenPlanerApp is the android app frontend for the KitchenPlaner, a tool
+ * to cooperatively plan a meal plan for a campout.
+ * Copyright (C) 2023  Arne Kuchenbecker, Antonia Heiming
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/exceptions/DuplicatePrimaryKeyException.kt:4]: Change Licence

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/daos/RecipeManagementDAO.kt [hunk 5]
```diff
@@ -0,0 +1,28 @@
+/*
+ * KitchenPlanerApp is the android app frontend for the KitchenPlaner, a tool
+ * to cooperatively plan a meal plan for a campout.
+ * Copyright (C) 2023  Arne Kuchenbecker, Antonia Heiming
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/datalayer/daos/RecipeManagementDAO.kt:4]: Change Licence

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/entities/RecipeProjectMealEntity.kt [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+@Entity(
+    tableName = "recipeProjectMeal",
+    primaryKeys = ["projectId", "meal"],
+    foreignKeys = [
+        ForeignKey(
+            entity = ProjectEntity::class,
+            parentColumns = ["id"],
+            childColumns = ["projectId"]
+        )/*,
+        ForeignKey(
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/datalayer/entities/RecipeProjectMealEntity.kt:32]: Change to TODO

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/repositories/ProjectRepository.kt [hunk 7]
```diff
@@ -55,7 +99,9 @@ class ProjectRepository @Inject constructor(
     }
 
     fun getAllProjectsOverview() : Flow<List<ProjectStub>> {
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/datalayer/repositories/ProjectRepository.kt:101]: Only for testing purposes

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/repositories/ProjectRepository.kt
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/datalayer/repositories/ProjectRepository.kt:1]: How to:
- add alternative recipe
- add picture
- change number of people
-change allergen persons
- get recipes for a project (to analyze the allergens)

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/ModelConversion.kt [hunk 8]
```diff
@@ -44,3 +47,30 @@ fun AllergenPerson.toDataLayerEntity(projectId: Long?) : Pair<AllergenPersonEnti
         AllergenEntity(projectId ?: 0, name, it.allergen, it.traces)
     })
 }
+
+fun ProjectEntity.toModelEntity(
+    meals: List<MealEntity>,
+    allergenPersons: List<AllergenPersonEntity>,
+    allergens: List<AllergenEntity>,
+    personNumbers: List<PersonNumberChangeEntity>
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/datalayer/ModelConversion.kt:55]: Unused

**Review by kittyahy — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/repositories/ProjectRepository.kt
Comment (arnekuchenbecker) [reply app/src/main/java/com/scouts/kitchenplaner/datalayer/repositories/ProjectRepository.kt:1]: Recipe related stuff should be done once recipes are available

**Review by kittyahy — APPROVED** [review]

**Pull request merged**